### ISSUE TITLE
Feature request: opt-in Assert<T> type for arguments

### ISSUE THREAD

**Issue by jedwards1211** [codemix/flow-runtime#222]
This is a:

 - [ ] Bug Report
 - [x] Feature Request
 - [ ] Question
 - [ ] Other

Which concerns:

 - [ ] flow-runtime
 - [x] babel-plugin-flow-runtime
 - [ ] flow-runtime-validators
 - [ ] flow-runtime-mobx
 - [ ] flow-config-parser
 - [ ] The documentation website

---

This is a PR I'd like to work on someday after fixing more important issues.

### What is the current behaviour?

I typically use `babel-plugin-flow-runtime` with `"optInOnly": true` for API argument validation.  The code looks like the following:

```js
// @flow
// @flow-runtime enable

import reify from 'flow-runtime'
import type {Type} from 'flow-runtime'

type InitUserAttributes = {
  email: string,
  password: string,
}
type InitUserAttributesType = (reify: Type<InitUserAttributes>)

export function createUser(attributes: InitUserAttributes) {
  InitUserAttributesType.assert(attributes)
  ...
}
```
---

### What is the expected behaviour?

It would be nice if I could shorten this to
```js
// @flow
// @flow-runtime enable

import type {Assert} from 'flow-runtime'

type InitUserAttributes = {
  email: string,
  password: string,
}

export function createUser(attributes: Assert<InitUserAttributes>) {
  ...
}
```

Basically `type Assert<T> = T`, but `babel-plugin-flow-runtime` would insert runtime assertions for any arguments using the `Assert` type.

### Which package versions are you using?

`babel-plugin-flow-runtime`: `0.18.0`
